datastream-server/src/main/java/com/linkedin/datastream/server/assignment/LoadBasedPartitionAssignmentStrategy.java

```diff
@@ -85,18 +94,19 @@ public LoadBasedPartitionAssignmentStrategy(PartitionThroughputProvider throughp
     Validate.isTrue(currentAssignment.size() > 0,
         "Zero tasks assigned. Retry leader partition assignment.");
 
+    // Resolving cluster name from datastream group
+    String clusterName = _sourceClusterResolver.getSourceCluster(datastreamPartitions.getDatastreamGroup());
+    ClusterThroughputInfo clusterThroughputInfo = partitionThroughputInfo.get(clusterName);
```

Comment:
Just wondering, if fetchPartitionThroughputInfo object should have cluster information. Since you have the retry logic at the earlier step (`fetchPartitionThroughputInfo`, the implementation for Brooklin-monitor/Pinot will be tricky, as you will need fabric/cluster information.

Reply:
`ClusterThroughputInfo` already has the `clusterName`. Let's take this offline, maybe I'm missing something.

Reply:
As per our discussion, this will go out in a separate PR.